dbt/adapters/athena/impl.py

```diff
@@ -203,55 +216,52 @@ def s3_table_location(
         if external_location and not is_temporary_table:
             return external_location.rstrip("/")
 
-        if not s3_path_table_part:
-            s3_path_table_part = table_name
+        s3_path_table_part = relation.s3_path_table_part or relation.identifier
+        schema_name = relation.schema
+        s3_data_naming = self._s3_data_naming(s3_data_naming)
+        table_prefix = self._s3_table_prefix(s3_data_dir)
 
         mapping = {
-            "uuid": path.join(self.s3_table_prefix(s3_data_dir), str(uuid4())),
-            "table": path.join(self.s3_table_prefix(s3_data_dir), s3_path_table_part),
-            "table_unique": path.join(self.s3_table_prefix(s3_data_dir), s3_path_table_part, str(uuid4())),
-            "schema_table": path.join(self.s3_table_prefix(s3_data_dir), schema_name, s3_path_table_part),
-            "schema_table_unique": path.join(
-                self.s3_table_prefix(s3_data_dir), schema_name, s3_path_table_part, str(uuid4())
-            ),
+            S3DataNaming.UUID: path.join(table_prefix, str(uuid4())),
+            S3DataNaming.TABLE: path.join(table_prefix, s3_path_table_part),
+            S3DataNaming.TABLE_UNIQUE: path.join(table_prefix, s3_path_table_part, str(uuid4())),
+            S3DataNaming.SCHEMA_TABLE: path.join(table_prefix, schema_name, s3_path_table_part),
+            S3DataNaming.SCHEMA_TABLE_UNIQUE: path.join(table_prefix, schema_name, s3_path_table_part, str(uuid4())),
         }
-        table_location = mapping.get(s3_data_naming)
 
-        if table_location is None:
-            raise ValueError(f"Unknown value for s3_data_naming: {s3_data_naming}")
-
-        return table_location
+        return mapping.get(s3_data_naming)
```

Comment:
shall we return `mapping[s3_data_naming]` then a ValueError is raised in case a malformed or wrong S3DataNaming is passed? doing so we avoid to deal with weird downstream None error for table location

Reply:
👍🏻